python/cudf/cudf/core/frame.py

```diff
@@ -3594,6 +3594,64 @@ def _colwise_binop(
 
         return output
 
+    def dot(self, other, reflect=False):
+        """
+        Get dot product of frame and other, (binary operator `dot`).
+
+        Among flexible wrappers (`add`, `sub`, `mul`, `div`, `mod`, `pow`,
+        `dot`) to arithmetic operators: `+`, `-`, `*`, `/`, `//`, `%`, `**`,
+        `@`.
+
+        Parameters
+        ----------
+        other : sequence, Series, or DataFrame
+            Any multiple element data structure, or list-like object.
+        reflect : bool, default False
+            If ``True`` the operation is reflected (i.e whether to swap the
+            left and right operands).
+
+        Returns
+        -------
+        scalar, Series, or DataFrame
+            The result of the operation.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame([[1, 2, 3, 4],
+        ...                      [5, 6, 7, 8]])
+        >>> df @ df.T
+            0    1
+        0  30   70
+        1  70  174
+        >>> s = cudf.Series([1, 1, 1, 1])
+        >>> df @ s
+        0    10
+        1    26
+        dtype: int64
+        >>> [1, 2, 3, 4] @ s
+        array(10)
```

Comment:
Taking the dot product of two vectors (1-dimensional cupy arrays) gives us a 0-d array - do we want to extract the value from this, and if so, is there a preferred way to do this?

Reply:
Looks like you can either convert to a specific type, eg int(x) or float(x) or  using x.item() where x =the 0-d array.
source: https://stackoverflow.com/questions/9814226/error-extracting-element-from-an-array-python